src/main/kotlin/com/hibob/academy/resource/OwnerResource.kt

```diff
@@ -0,0 +1,57 @@
+package com.hibob.academy.resource
+
+import com.hibob.academy.entity.Owner
+import jakarta.ws.rs.*
+import jakarta.ws.rs.core.MediaType
+import jakarta.ws.rs.core.Response
+import jakarta.ws.rs.core.Response.Status
+import org.springframework.stereotype.Controller
+
+@Controller
+@Path("/api/owner")
+@Produces(MediaType.APPLICATION_JSON)
+class OwnerResource {
+
+    @GET
+    fun getALLOwners() : Response {
+        return Response.ok(listOf(
+            Owner(1, "Shahar", 1, 1, null,null),
+            Owner(2, "Alex", 1, 2, null,null )
+        ))
+            .build()
+    }
+
+    @POST
+    @Consumes(MediaType.APPLICATION_JSON)
+    fun createOwner(owner: Owner): Response {
+
+        owner.name = owner.name ?: run {
+            if (owner.firstName.isNullOrEmpty() || owner.lastName.isNullOrEmpty()) {
+                return Response.status(Status.BAD_REQUEST).entity("Name is required").build()
```

Comment:
Nice! You can also just throw a BadInputException("Name is required"). The framework catches the BadInputException and handles the Response.status for you.